test/e2e/storage/testsuites/subpath.go

```diff
@@ -0,0 +1,755 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
+	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	volumePath      = "/test-volume"
+	volumeName      = "test-volume"
+	probeVolumePath = "/probe-volume"
+	probeFilePath   = probeVolumePath + "/probe-file"
+	fileName        = "test-file"
+	retryDuration   = 10
+	mountImage      = imageutils.GetE2EImage(imageutils.Mounttest)
+)
+
+type subPathTestSuite struct {
```

Comment:
for each of the types in your change, could you explicitly specify which interfaces it implements. For example:
`var _ TestSuite = &subPathTestSuite{}`
This forces compile time checking of interface implementation and makes it explicit for any reader to know what interfaces this type implements

Reply:
Fixed as suggested. PTAL